src/flask/ctx.py

```diff
@@ -130,7 +130,14 @@ def add_header(response):
 
     .. versionadded:: 0.9
     """
-    _request_ctx_stack.top._after_request_functions.append(f)
+    try:
+        _request_ctx_stack.top._after_request_functions.append(f)
+    except Exception:
```

Comment:
Do not catch all exceptions wildly (and neither just swallow them, it should remain an exception). I pointed out code in another function that does the right thing - simply do the same here.